### PR TITLE
#166789877  Users should be able to fetch article's ratings

### PR DESCRIPTION
**Pull request by manzif**
#### What does this PR do?

 - Users should be able to fetch article's ratings

#### Description of Task to be completed?

- fetch ratings of a given article
- get an average of ratings.
- get the number of users who rated the article

#### How should this be manually tested?

- When you login in the system and rate an article, Every user in the system can view ratings of any article of their choice. 

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [#166789877](https://www.pivotaltracker.com/story/show/166789877)

#### Screenshots (if appropriate)

-image

-image

#### Questions:

- N/A

### CODE REVIEW

File: src/db/migrations/20190710092805-create-draftarticle.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      updatedAt: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  down: (queryInterface, Sequelize) => {
+    return queryInterface.dropTable('draftarticles');
+  }
+};
```
Comment (j4l13n) [inline src/db/migrations/20190710092805-create-draftarticle.js]: - can you remove new line issues on your files

**Review by j4l13n — APPROVED** [review]

**Review by adafia — APPROVED** [review]

**Comment by Emile-Nsengimana** [comment]
> Article Not found or it has not rated

I think the response message above when an article is not found can be improved

File: src/routes/api/draftarticles.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// import express from 'express';
```
Comment (Dom58) [inline src/routes/api/draftarticles.js]: Why those useless comments and not JDOC!?

Comment (manzif) [reply src/routes/api/draftarticles.js]: Resolved

**Comment by manzif** [comment]
> > Article Not found or it has not rated
> 
> I think the response message above when an article is not found can be improved

Please take a look at the new message. Thank you

**Review by Dom58 — APPROVED** [review]

File: tests/ratesTest.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import Chai from 'chai';
```
Comment (emukungu) [inline tests/ratesTest.js]: change 'Chai' to lowercase

File: src/routes/api/draftratings.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// import express from 'express';
```
Comment (emukungu) [inline src/routes/api/draftratings.js]: Remove all commented out code

File: src/index.js [hunk 5]
```diff
@@ -32,8 +32,8 @@ app.use(session({
 app.use(passport.initialize());
 app.use(passport.session());
 app.use(routes);
-// cron from deleting blacklist tokens
-cron.schedule('* * * * *', () => {
+// cron fro deleting blacklist tokens
```
Comment (emukungu) [inline src/index.js]: manage your grammar properly. remove commented out code

File: src/controllers/articleRatingControllers.js [hunk 6]
```diff
@@ -63,5 +63,36 @@ class ArticleRatingManager {
...
+      attributes: ['rating'],
+      raw: true,
+    });
+    if (averageRating.length === 0) {
+      return res.status(400).send({
+        message: 'This Article is Not found or it has not been rated yet,Please try agai'
+      });
+    }
+    let totalrating = 0;
+    // eslint-disable-next-line array-callback-return
```
Comment (emukungu) [inline src/controllers/articleRatingControllers.js]: you don't need to ignore this line; fix the eslint issue

File: tests/ratesTest.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+Chai.use(chaiHttp);
+Chai.should();
+Chai.expect();
+dotenv.config();
+
+
+describe('Fetch Rating Routes', () => {
+  before('should be able to register a new rate', async () => {
+    const res = await Chai.request(index).post('/api/users/login').set('Content-Type', 'application/json').send({
```
Comment (emukungu) [inline tests/ratesTest.js]: improve this line readability, structure your code better.

**Review by emukungu — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 5 above)
Comment (manzif) [reply src/index.js]: On this side, I don't find commented out code??

File: src/routes/api/index.js [hunk 8]
```diff
@@ -1,13 +1,15 @@
 import express from 'express';
 import users from './users';
 import profiles from './profiles';
+// import ratingsstuff from './draftratings';
```
Comment (emukungu) [inline src/routes/api/index.js]: this is commented out code

File: src/routes/api/index.js [hunk 9]
```diff
@@ -1,13 +1,15 @@
...
 import articles from './articles';
 import auth from './auth';
 
 const router = express.Router();
 
 router.use('/api', users);
 router.use('/api', profiles);
+// router.use('/api', ratingsstuff);
```
Comment (emukungu) [inline src/routes/api/index.js]: this is commented out code

**Review by emukungu — CHANGES_REQUESTED** [review]

File: src/routes/api/index.js (hunk 8 above)
Comment (manzif) [reply src/routes/api/index.js]: Resplved

File: src/routes/api/index.js (hunk 9 above)
Comment (manzif) [reply src/routes/api/index.js]: Resolved

File: src/db/models/draftrating.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
```
Comment (emukungu) [inline src/db/models/draftrating.js]: why do you have a draftrating model. This is mock data

File: src/controllers/articleRatingControllers.js [hunk 11]
```diff
@@ -63,5 +63,36 @@ class ArticleRatingManager {
...
+ */
+  static async ratingAverage(req, res) {
+    const averageRating = await Rating.findAll({
+      where: { articleId: req.params.articleId },
+      attributes: ['rating'],
+      raw: true,
+    });
+    if (averageRating.length === 0) {
+      return res.status(400).send({
+        message: 'This Article is Not found or it has not been rated yet,Please try agai'
```
Comment (emukungu) [inline src/controllers/articleRatingControllers.js]: The message should only reflect the article not having been rated. 'Try again' is not necessary

File: src/controllers/articleRatingControllers.js [hunk 12]
```diff
@@ -63,5 +63,36 @@ class ArticleRatingManager {
...
+ * @param {object} res
+ * @returns {message} That article is not available
+ */
+  static async ratingAverage(req, res) {
+    const averageRating = await Rating.findAll({
+      where: { articleId: req.params.articleId },
+      attributes: ['rating'],
+      raw: true,
+    });
+    if (averageRating.length === 0) {
```
Comment (emukungu) [inline src/controllers/articleRatingControllers.js]: change 'averageRating.length === 0' to es6

**Review by emukungu — CHANGES_REQUESTED** [review]

**Pull request merged**